carAPI/src/main/java/com/everis/boundary/CarResource.java

```diff
@@ -63,31 +52,31 @@ public Response getCar(@PathParam("carId") long id) {
 	//Recibe un coche y lo a�ade al array
 	@POST
 	public Response createCar(Car car, @Context UriInfo uriInfo) {
-		cars.put(car.getId(), car);
-		String newId = String.valueOf(car.getId());
+		Car newCar = carService.createCar(car);
+		String newId = String.valueOf(newCar.getId());
 		URI uri = uriInfo.getAbsolutePathBuilder().path(newId).build();
-		return Response.created(uri).entity(car).build();
+		return Response.created(uri).entity(newCar).build();
 	}
 
 	//Recibe un coche y actualiza sus datos
 	@PUT
 	@Path("/{carId}")
 	public Response updateCar(@PathParam("carId") long id, Car car) {
 		car.setId(id);
-		if(car.getId()<=0) {
-			//Devolver mediante objeto error
+		Car newCar = carService.updateCar(car);
+		if(newCar==null) {
 			return Response.status(Status.NOT_FOUND).build();
 		}
-		cars.put(car.getId(), car);
-		return Response.status(Status.OK).entity(car).build();
+		else {
+			return Response.status(Status.OK).entity(newCar).build();
+		}
 	}
 
 	//Recibe un id de coche y borra el coche con ese id
 	@DELETE
 	@Path("/{carId}")
 	public Response deleteCar(@PathParam("carId") long id) {
-		Car newCar = cars.remove(id);
-		//A�adir logica
+		Car newCar = carService.deleteCar(id);
```

Comment:
If we are deleting a car maybe a variable named "newCar" does not really fit here 😄